### ISSUE TITLE
URLs compromised under Windows

### ISSUE THREAD

**Issue by bneumann** [zettajs/zetta#331]
Following workflow:

checkout helloworld git repo:
https://github.com/zettajs/zetta-hello-world

Build and run under Windows 10. Goto http://localhost:1337 Following json appears:

`{
rel: [
"self",
"edit"
],
href: "http://localhost:1337/\servers\FirstName%20LastName\devices\9e967da5-bac2-4fff-b2ff-3541385ccf51"
},
{
rel: [
"http://rels.zettajs.io/type",
"describedby"
],
href: "http://localhost:1337/\servers\FirstName%20LastName/meta/state_machine"
},
{
title: "FirstName LastName",
rel: [
"up",
"http://rels.zettajs.io/server"
],
href: "http://localhost:1337/\servers\FirstName%20LastName"
}`

If I run the same under linux I revceive correct URLS:

http://localhost:1337/servers/FirstName%20LastName

Problem is, that the REST API is broken without correct links and the Zetta Browser does not show anything

**Comment by mdobson** [comment]
Hey @bneumann thanks for reporting this. What version of zetta are you using with that tutorial?

**Comment by mdobson** [comment]
Hey @bneumann just looked at the repo you refer to, and it was using very old versions of zetta and the mock packages. I've updated them to latest. Can you let me know if that changes anything for you? I don't have a windows system to test with currently.

**Comment by bneumann** [comment]
Hi, 
I used the "latest" version of zetta in my package file. Should be 1.3.2 now.
The URLs are still not accessible

**Comment by mdobson** [comment]
@AdamMagaluk Have access to a windows box? Would you be able to make a pass at a fix?

**Comment by bneumann** [comment]
I would fix it if you could push me in the right direction. I browsed trhough the zetta code and did not find the correct file where the magic formatting happens

**Comment by kevinswiber** [comment]
@AdamMagaluk @mdobson It would help to look for any `path.join` instances that alter the URL but don't include a replacement for Windows.  From https://github.com/argo/argo-url-helper/blob/5e5cd602eee51651cb4b47c5d9208f213d8c4b75/url.js#L29:

```js
parsed.pathname = path.join(parsed.pathname, pathname).replace(/\\/g, '/');
```

**Comment by mdobson** [comment]
Sure. I'd be more than happy to offer some guidance.

The API server underlying zetta is a package called `argo`. It manages all the http transactions, and does the nifty formatting stuff. It also has a concept of middleware similar to connect middleware in express. 

The `argo-url-helper` module is where we do the url formatting. 

As @kevinswiber mentioned above a good starting point is this line. https://github.com/argo/argo-url-helper/blob/master/url.js#L29

**Comment by kevinswiber** [comment]
Here's a case where it will fail:

https://github.com/zettajs/zetta/blob/d3b3727eec6d95913e6b4eeddfd8bebff1d4689c/lib/http_server.js#L276

**Comment by kevinswiber** [comment]
A couple more:

* https://github.com/argo/argo-url-helper/blob/5e5cd602eee51651cb4b47c5d9208f213d8c4b75/url.js#L93
* https://github.com/argo/argo-url-helper/blob/5e5cd602eee51651cb4b47c5d9208f213d8c4b75/url.js#L112

**Comment by bneumann** [comment]
@kevinswiber Ah so I was already in the right spot. But tbh I was a bit confused what these modules do. But thanks to the explanation of @AdamMagaluk that it is a middlewary kind of thing I can take a deeper look into it. I let you know what I find out. Thanks for the tips!

**Comment by mdobson** [comment]
Great! Let us know if you have any other questions.

**Comment by bneumann** [comment]
Okay I made 2 pull requests on the two repositories. It is currently building, but it works now on my machine. Thanks guys!

**Comment by AdamMagaluk** [comment]
Fixed with #332 new release coming.

**Comment by AdamMagaluk** [comment]
@bneumann It's fixed in zetta@1.4.0
